src/hotspot/os/windows/os_windows.cpp

```diff
@@ -862,35 +862,38 @@ bool os::win32::available_memory(size_t& value) {
   MEMORYSTATUSEX ms;
   ms.dwLength = sizeof(ms);
   BOOL res = GlobalMemoryStatusEx(&ms);
-  if (res) {
+  if (res == TRUE) {
     value = static_cast<size_t>(ms.ullAvailPhys);
     return true;
   } else {
-    return false;
+    assert(false, "GlobalMemoryStatusEx failed in win32::available_memory().");
+    return res == TRUE;
```

Comment:
```suggestion
    return false;
```
Please restore all these to "return false" as we know that `res == false` here.

Reply:
Addressed.